src/main/java/com/flightstats/hub/channel/ChannelResource.java

```diff
@@ -275,4 +283,13 @@ public Response delete(@PathParam("channel") final String channelName) throws Ex
         log.info("using normal delete {}", channelName);
         return deletion(channelName);
     }
+
+    private void checkPermission(String task, String name) {
```

Comment:
My preference is still to create a guicey/aop @Forbidden(mode=ReadOnly) annotation processor to handle this, if we want to start using guice aop.  Pulled this out to a helper method for now.

Reply:
I know AOP conceptually, but I seems like a reasonable candidate for that type of thing. I have very little experience with actual AOP implementation though. 

The one time I used it left a bad taste in my mouth because it made attaching a debugger very messy...I couldn't set a breakpoint on the method invoking the AOP behavior that I wanted, I had to set it on the AOP code which got invoked in dozens of places and thus fired the debugger dozens of times I didn't care about.

And, depending on if it's code generation or byte code manipulation, it can screw with the line numbers in the file making them not match the line numbers in your code, making it really painful to line up stack traces.

AOP was cool in theory, but in practice it was painful. Again...just had that one experience long ago, so it might not be relevant now.

Reply:
Yeah, you can definitely get in trouble with it.  I've found it cleans up code quite a bit, if used judiciously in clear ways and the team is familiar with it.  Guice's version plays well with other guicey bits.
https://github.com/google/guice/wiki/AOP